suit_generator/cmd_cache_create.py

```diff
@@ -6,116 +6,297 @@
-    """
-    Create a raw SUIT DFU cache file from input binaries.
-
-    This function processes a list of input binaries, each associated with a URI, and creates a raw SUIT DFU cache file.
-    The data is padded to align with the specified erase block size (eb_size).
-
-    :param input: List of input binaries with corresponding URIs, passed in the format <URI>,<INPUT_FILE_PATH>.
-    :type input: list[str]
-    :param output_file: Path to the output raw SUIT DFU cache file.
-    :type output_file: str
-    :param eb_size: Erase block size in bytes (used for padding).
-    :type eb_size: int
-    :return: None
-    """
-    cache_data = bytes()
-    first_slot = True
 
-    for single_input in input:
-        args = single_input.split(",")
-        if len(args) < 2:
-            raise ValueError("Invalid number of input arguments: " + single_input)
-        uri, input_file = args
+class CachePartition:
+    def __init__(self, eb_size: int):
+        self.first_slot = True
+        self.cache_data = bytes()
+        self.eb_size = eb_size
 
-        data = None
-        with open(input_file, "rb") as f:
-            data = f.read()
+    def add_padding(self, data: bytes) -> bytes:
+        """
+        Add padding to the given data to align it to the specified erase block size.
+
+        This method ensures that the data is padded to a size that is a multiple of the erase block size (self.eb_size).
+        The padding is done by appending a CBOR key-value pair with empty URI as the key and
+        byte-string-encoded zeros as the value.
+
+        :param data: The input data to be padded.
+        :type data: bytes
+        :return: The padded data.
+        """
+        rounded_up_size = math.ceil(len(data) / self.eb_size) * self.eb_size
+        padding_size = rounded_up_size - len(data)
+        padded_data = data
+
+        # minimum padding size is 2 bytes
+        if padding_size == 1:
+            padding_size += self.eb_size
+            rounded_up_size += self.eb_size
+
+        if padding_size == 0:
+            return data
+
+        padded_data += bytes([0x60])
+
+        if padding_size <= 23:
+            header_len = 2
+            padded_data += bytes([0x40 + (padding_size - header_len)])
+        elif padding_size <= 0xFFFF:
+            header_len = 4
+            padded_data += bytes([0x59]) + (padding_size - header_len).to_bytes(2, byteorder="big")
+        else:
+            raise ValueError("Number of required padding bytes exceeds assumed max size 0xFFFF")
+
+        return padded_data.ljust(rounded_up_size, b"\x00")
 
+    def add_cache_slot(self, uri: str, data: bytes):
+        """
+        Add a cache slot to the cache from the given URI and data.
+
+        This function creates a cache slot from the given URI and data, and pads the data to align with the specified
+        erase block size (eb_size). The first slot in the cache is created with indefinite length CBOR map.
```

Comment:
As discussed - the end of map marker 0xFF must be placed in a new erase block.

If the device itself would try to add a new cache slot it would otherwise face a problem - the current end of map marker would have to be removed. This would cause the need to erase the whole previous erase block with data from the previous slot.
Though we might need to waste one erase block (at the end of the cache), this is unavoidable.